sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/GetTypeWithIsAssignableFrom.cs

```diff
@@ -129,13 +138,14 @@ private static void CheckIfIsExpressionCanBeReplacedByNullCheck(SyntaxNodeAnalys
         private static void CheckGetTypeAndTypeOfEquality(ExpressionSyntax sideA, ExpressionSyntax sideB, Location location,
             SyntaxNodeAnalysisContext context)
         {
-            if (!TypeExaminationOnSystemType.IsGetTypeCall(sideA as InvocationExpressionSyntax, context.SemanticModel))
+            var typeSyntax = (sideB as TypeOfExpressionSyntax)?.Type;
+            if (typeSyntax == null)
             {
                 return;
             }
 
-            var typeSyntax = (sideB as TypeOfExpressionSyntax)?.Type;
-            if (typeSyntax == null)
+            if (!sideA.ToString().Contains("GetType") ||
```

Comment:
I think we can move this up since checking for string is cheap and we can do this before retrieving the `sideB` type.